pkgs/applications/editors/emacs/head.nix

```diff
@@ -0,0 +1,160 @@
+{ stdenv, lib, fetchurl, ncurses, xlibsWrapper, libXaw, libXpm
+, Xaw3d, libXcursor,  pkgconfig, gettext, libXft, dbus, libpng, libjpeg, libungif
+, libtiff, librsvg, gconf, libxml2, imagemagick, gnutls, libselinux
+, alsaLib, cairo, acl, gpm, AppKit, GSS, ImageIO, m17n_lib, libotf
+, fetchFromGitHub
+, systemd ? null
+, withX ? !stdenv.isDarwin
+, withNS ? stdenv.isDarwin
+, withGTK2 ? false, gtk2-x11 ? null
+, withGTK3 ? true, gtk3-x11 ? null, gsettings-desktop-schemas ? null
+, withXwidgets ? false, webkitgtk ? null, wrapGAppsHook ? null
+, withCsrc ? true
+, srcRepo ? true, autoconf ? null, automake ? null, texinfo ? null
+, siteStart ? ./site-start.el
+}:
+
+assert (libXft != null) -> libpng != null;      # probably a bug
+assert stdenv.isDarwin -> libXaw != null;       # fails to link otherwise
+assert withNS -> !withX;
+assert withNS -> stdenv.isDarwin;
+assert (withGTK2 && !withNS) -> withX;
+assert (withGTK3 && !withNS) -> withX;
+assert withGTK2 -> !withGTK3 && gtk2-x11 != null;
+assert withGTK3 -> !withGTK2 && gtk3-x11 != null;
+assert withXwidgets -> withGTK3 && webkitgtk != null;
+
+let
+  toolkit =
+    if withGTK2 then "gtk2"
+    else if withGTK3 then "gtk3"
+    else "lucid";
+in
+stdenv.mkDerivation rec {
+  name = "emacs-${version}${versionModifier}";
+  version = "27";
+  versionModifier = "";
+
+  src = fetchFromGitHub {
+    owner = "emacs-mirror";
+    repo = "emacs";
+    rev = "dfb0ba79b5f41ca6fed25a03d2a5cd6996ec4753";
+    sha256 = "0cr8mlcxyib4rd8xj65hb1wxwlmppgg4823z7h3dbbxi273gz5b0";
+  };
+
+  enableParallelBuilding = true;
+
+  patches = [
+    # ./clean-env.patch
+    ./tramp-detect-wrapped-gvfsd.patch
+  ];
+
+  postPatch = lib.optionalString srcRepo ''
+    rm -fr .git
+  '';
+
+  CFLAGS = "-DMAC_OS_X_VERSION_MAX_ALLOWED=101200";
+
+  nativeBuildInputs = [ pkgconfig ]
+    ++ lib.optionals srcRepo [ autoconf automake texinfo ]
+    ++ lib.optional (withX && (withGTK3 || withXwidgets)) wrapGAppsHook;
+
+  buildInputs =
+    [ ncurses gconf libxml2 gnutls alsaLib acl gpm gettext ]
```

Comment:
Based off of the release notes, it looks like `gmp` and `jansson` should be added to `buildInputs`.
- `gmp`: The build now requires gmp, but will fallback to using the included `minigmp` if it's not available. I'm thinking it would be better to use `gmp` from nixpkgs.
- `jansson`: Emacs now uses `jansson` by default if it's available, for faster JSON parsing.

Reply:
thanks! i've made this change